src/librustc/hir/mod.rs

```diff
@@ -389,24 +389,33 @@ impl PathSegment {
     }
 }
 
+#[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
+pub struct ConstArg {
+    pub value: AnonConst,
+    pub span: Span,
```

Comment:
This span is identical to the span in body of `AnonConst`, is it viable to use it instead?
Other `AnonConst`s don't create extra spans during lowering as far as can I see.

Reply:
(The whole `ConstArg` could be refactored away in that case.)

Reply:
Ugh, I see `GenericArg::span` wants to work in isolation without any contexts.

Reply:
Yeah, we went through the same thought process when this was implemented. If you can see a way to refactor this, I'd be all ears.